src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -182,9 +214,13 @@ private void InstallWorkloadsWithInstallRecord(IEnumerable<WorkloadId> workloadI
                         {
                             _workloadInstaller.WriteWorkloadInstallationRecord(workloadId, sdkFeatureBand);
                         }
-
                     },
                     rollback: () => {
+                        foreach (var manifest in manifestsToUpdate)
+                        {
+                            _workloadInstaller.InstallWorkloadManifest(manifest.manifestId, manifest.existingVersion, sdkFeatureBand);
```

Comment:
check the doc of TransactionalAction. _workloadInstaller.InstallWorkloadManifest has a layer of rollback already. You should not need to say it here

Reply:
or write a small program to try out